packages/protocol/contracts/common/UsingRegistryV2.sol

```diff
@@ -0,0 +1,119 @@
+  bytes32 constant GOVERNANCE_SLASHER_REGISTRY_ID = keccak256(
+    abi.encodePacked("GovernanceSlasher")
+  );
+  bytes32 constant LOCKED_GOLD_REGISTRY_ID = keccak256(abi.encodePacked("LockedGold"));
+  bytes32 constant RESERVE_REGISTRY_ID = keccak256(abi.encodePacked("Reserve"));
+  bytes32 constant RANDOM_REGISTRY_ID = keccak256(abi.encodePacked("Random"));
+  bytes32 constant SORTED_ORACLES_REGISTRY_ID = keccak256(abi.encodePacked("SortedOracles"));
+  bytes32 constant STABLE_TOKEN_REGISTRY_ID = keccak256(abi.encodePacked("StableToken"));
+  bytes32 constant VALIDATORS_REGISTRY_ID = keccak256(abi.encodePacked("Validators"));
+  // solhint-enable state-visibility
+
+  modifier onlyRegisteredContract(bytes32 identifierHash) {
+    require(registry.getAddressForOrDie(identifierHash) == msg.sender, "only registered contract");
+    _;
+  }
+
+  modifier onlyRegisteredContracts(bytes32[] memory identifierHashes) {
+    require(registry.isOneOf(identifierHashes, msg.sender), "only registered contracts");
+    _;
+  }
+
+  function getAccounts() internal view returns (IAccounts) {
+    return IAccounts(registry.getAddressForOrDie(ACCOUNTS_REGISTRY_ID));
+  }
+
+  function getAttestations() internal view returns (IAttestations) {
+    return IAttestations(registry.getAddressForOrDie(ATTESTATIONS_REGISTRY_ID));
+  }
+
+  function getElection() internal view returns (IElection) {
+    return IElection(registry.getAddressForOrDie(ELECTION_REGISTRY_ID));
+  }
+
+  function getExchange() internal view returns (IExchange) {
+    return IExchange(registry.getAddressForOrDie(EXCHANGE_REGISTRY_ID));
+  }
+
+  function getFeeCurrencyWhitelistRegistry() internal view returns (IFeeCurrencyWhitelist) {
+    return IFeeCurrencyWhitelist(registry.getAddressForOrDie(FEE_CURRENCY_WHITELIST_REGISTRY_ID));
+  }
+
+  function getFreezer() internal view returns (IFreezer) {
+    return IFreezer(registry.getAddressForOrDie(FREEZER_REGISTRY_ID));
+  }
+
+  function getGoldToken() internal view returns (IERC20) {
+    return IERC20(registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID));
+  }
+
+  function getGovernance() internal view returns (IGovernance) {
+    return IGovernance(registry.getAddressForOrDie(GOVERNANCE_REGISTRY_ID));
+  }
+
+  function getLockedGold() internal view returns (ILockedGold) {
+    return ILockedGold(registry.getAddressForOrDie(LOCKED_GOLD_REGISTRY_ID));
+  }
+
+  function getRandom() internal view returns (IRandom) {
+    return IRandom(registry.getAddressForOrDie(RANDOM_REGISTRY_ID));
+  }
+
+  function getReserve() internal view returns (IReserve) {
+    return IReserve(registry.getAddressForOrDie(RESERVE_REGISTRY_ID));
+  }
+
+  function getSortedOracles() internal view returns (ISortedOracles) {
+    return ISortedOracles(registry.getAddressForOrDie(SORTED_ORACLES_REGISTRY_ID));
+  }
+
+  function getStableToken() internal view returns (IStableToken) {
+    return IStableToken(registry.getAddressForOrDie(STABLE_TOKEN_REGISTRY_ID));
+  }
```

Comment:
we now have a celo euro which should also be included here
I don't think we should deprecate the `getStableToken` but maybe we can provide an additional `getStableDollarToken` and `getStableEuroToken`

Reply:
the same goes for `Exchange`

Reply:
@yorhodes you mean we need to add get methods for Dollar and Euro? ex. getDollarExchange.
Also, should I need to create different interfaces that extend form IStableToken for Dollar and Euro?

Reply:
its the same interfave IStableToken and IExchange
just different identifiers